kale/embed/mpca.py

```diff
@@ -23,249 +21,251 @@
+        logging.error(error_msg)
+        raise ValueError(error_msg)
 
 
-def _check_dim_shape(X, ndim, shape_):
-    """Check if the order/mode of the input data is consistent with the given number of order/mode
+def _check_dim_shape(x, n_dim, shape_):
+    """Check whether the order of the input data is consistent with the given value of order
 
     Args:
-        X (ndarray): input data
-        ndim (int): number of order/mode expected
+        x (array-like): input data
+        n_dim (int): tensor order expected
         shape_: shape expected
 
     """
-    if not check_ndim(X, ndim) or not check_shape(X, shape_):
-        logging.error("The given data should be consistent with the order and shape of training data")
-        sys.exit()
+    _check_ndim(x, n_dim)
+    _check_shape(x, shape_)
 
 
 class MPCA(BaseEstimator, TransformerMixin):
     """MPCA implementation compatible with sickit-learn
 
     Args:
-        var_ratio (float, optional): Percentage of variance explained
-            (between 0 and 1). Defaults to 0.97.
+        var_ratio (float, optional): Percentage of variance explained (between 0 and 1). Defaults to 0.97.
         max_iter (int, optional): Maximum number of iteration. Defaults to 1.
-        vectorise (bool): Whether vectorise the transformed tensor. Defaults to Flase.
-        n_components (int): Number of components to keep. Applies only when vectorise=True.
-            Defaults to None.
+        return_vector (bool): Whether ruturn the transformed/projected tensor in vector. Defaults to False.
+        n_components (int): Number of components to keep. Applies only when return_vector=True. Defaults to None.
 
     Attributes:
-        proj_mats (list): A list of transposed projection matrices.
-        idx_order (ndarray): The ordering index of projected (and vectorised) features in decreasing variance.
-        mean_ (ndarray): Per-feature empirical mean, estimated from the training set, shape (I_1, I_2, ..., I_N).
+        proj_mats (list of array-like): A list of transposed projection matrices, shapes (P_1, I_1), ...,
+            (P_N, I_N), where P_1, ..., P_N are output tensor shape for each sample.
+        idx_order (array-like): The ordering index of projected (and vectorised) features in decreasing variance.
+        mean_ (array-like): Per-feature empirical mean, estimated from the training set, shape (I_1, I_2, ..., I_N).
         shape_in (tuple): Input tensor shapes, i.e. (I_1, I_2, ..., I_N).
         shape_out (tuple): Output tensor shapes, i.e. (P_1, P_2, ..., P_N).
     Examples:
         >>> import numpy as np
-        >>> from kale.embed.mpca import MPCA
-        >>> X = np.random.random((20, 25, 20, 40))
-        >>> X.shape
-        (20, 25, 20, 40)
+        >>> from kale.embed import MPCA
+        >>> x = np.random.random((40, 20, 25, 20))
+        >>> x.shape
+        (40, 20, 25, 20)
```

Comment:
This will be good for `tests`, right? Consider to use it. See #57

Reply:
Yeah, it is one option. I am also considering using the gait data for testing and toy example.